File: tampermonkey.js
// ==UserScript==
// @name         crack the code
// @namespace    codebat_raymond
// @version      0.1
// @description  crack the cookie
// @author       Raymond
// @match        http://www.toutiao.com/*
// @grant        none
// @require      https://sekiro.virjar.com/sekiro-doc/assets/sekiro_web_client.js
// ==/UserScript==

(function() {
    'use strict';
    function guid() {
    function S4() {
        return (((1+Math.random())*0x10000)|0).toString(16).substring(1);
    }
    return (S4()+S4()+"-"+S4()+"-"+S4()+"-"+S4()+"-"+S4()+S4()+S4());
    }
    var client = new SekiroClient("ws://127.0.0.1:5620/business-demo/register?group=tt-test&clientId="+guid());

    client.registerAction("toutiao",function(request, resolve, reject){

        var oss = request['url']
        resolve(window.byted_acrawler.sign({'url':oss}));
    })

})();

File: tampermonkey.py
# -*- coding:utf-8 -*-
# File Name: tampermonkey.py
# Author: Codebat_Raymond
# Date: 5/27/2023

import requests
data = {
    "group": "tt-test",
    "action": "toutiao",
    'url':'https://www.toutiao.com/api/pc/list/feed?offset=0&channel_id=94349549395&max_behot_time=0&category=pc_profile_channel&disable_raw_data=true&aid=24&app_name=toutiao_web'
    }
sign = requests.get("http://127.0.0.1:5620/business-demo/invoke",params=data )
_signature = sign.json().get('data')

url = 'https://www.toutiao.com/api/pc/list/feed?offset=0&channel_id=94349549395&max_behot_time=0&category=pc_profile_channel&disable_raw_data=true&aid=24&app_name=toutiao_web&_signature={}'.format(_signature)

print(url)
